src/backend/distributed/planner/recursive_planning.c

```diff
@@ -533,6 +1100,22 @@ ShouldRecursivelyPlanSubquery(Query *subquery)
 	}
 	else if (DeferErrorIfCannotPushdownSubquery(subquery, false) == NULL)
 	{
+		/*
+		 * We should do one more check for the distribution key equality.
+		 *
+		 * If the input query to the planner doesn't contain distribution key equality,
+		 * we should further check whether this individual subquery contains or not.
+		 *
+		 * If all the relations are not joined on their distribution keys for the
```

Comment:
If all the relations are not joined -> if not all relations are joined

(the error message that says this always bugs me :) )